server/resources/resource_management.go

```diff
@@ -1377,17 +1388,35 @@ func validateAttachmentRelation(
 	return nil
 }
 
+func checkRelatedResource(relatedResource *schema.Resource, data map[string]interface{}, policy *schema.Policy) bool {
+	rc := policy.GetOtherResourceCondition()
+	if rc.SkipTenantDomainCheck() {
+		return true
+	}
+	return equalStringKeys(data, relatedResource.Data(), tenantIDKey) &&
+		equalStringKeys(data, relatedResource.Data(), domainIDKey)
+}
+
+func equalStringKeys(first, second map[string]interface{}, key string) bool {
+	if firstValue, ok := first[key].(string); ok {
+		if secondValue, ok := second[key].(string); ok {
+			return firstValue == secondValue
+		}
+	}
+	return true
```

Comment:
I am not sure but double check that in case when resource is not related to nor tenant nor domain there is no key tenant_in/domain_id in this dict. I think it can contain just empty strings in this case.

Reply:
tenant_id/domain_id are relation to tenant schema so they can be either proper value of null - in the second case they won't be key in data